[IOSP-414] CRP Bot: Reuse existing Jira version if already existing

AliSoftware opened the pull request:
Ticket: https://babylonpartners.atlassian.net/browse/IOSP-414

### Why?

Some teams and PMs create the Version at the beginning of the sprint or during the sprint, essentially to forecast which tickets they plan to put in said version even before the release branch is cut.

This means that by the time the `/crp` command is run in Slack, some JIRA Versions already exist with the same version name that the bot wants to use

### How?

For each board, before creating a new version, first fetch the list of existing JIRA versions of the board and if there's one already existing with the proper `name`, reuse it. Only create a new version if there is none already existing with the expected name.

### PR checklist

* [x] I've assigned this PR to myself

With :heart:

Commit:
Checking if version exists before creating new one

Commit:
Fix tests

ilyapuchka commented on Sources/Stevenson/JiraService.swift:
```diff
@@ -130,24 +130,51 @@ extension JiraService {
+        self.logger.info("[JIRA] \(logMessage)")
+
+        return try container.make(SlowClient.self)
+                .get(fullURL, headers: self.headers, on: container) { request in
```
some extra indentation here

ilyapuchka commented on Sources/Stevenson/JiraService.swift:
```diff
@@ -130,24 +130,51 @@ extension JiraService {
+
+        let projectKey = self.knownProjects.first(where: { $0.value == projectID })?.key ?? "#\(projectID)"
+        let logMessage = "Fetching JIRA versions for board <\(projectKey)>"
+        self.logger.info("[JIRA] \(logMessage)")
```
off-topic: could have a logger helper method to take care of prefix

AliSoftware replied:
Fixed, indeed feels more in par with the other helpers already existing in this file anyway 👍

Commit:
Fix indentation

Commit:
Merge branch 'master' into olivier/IOSP-414

Commit:
Use FailableService.request method to wrap parsing

And avoid repeating the `decode(ServiceError.self).thenThrowing { throw $0 }` for every parsing in JIRA requests
